### ISSUE TITLE
Multiple Identity Servers - Same Replicated Backend

### ISSUE THREAD

**Issue by Wayne-Mather** [IdentityServer/IdentityServer4#868]
Hi,

**TLDR;**

o Can an identity server font-end and back-end be replicated like ASP.NET Cookie Authentication?

**Full Version**

I have an ASP.NET MVC Core website (1.1.0) that is distributed into remote areas with a replicated SQL Server back-end. Due to reasons I also need the head office server to provide front-ends for fall back data entry if the remote sites are unable to work for any reason.

Due to the way the application is configured I need to run separate web front ends for each of the servers. I would like to implemented identity server 4 to allow SSO across each site in the head office.

Before starting this journey, is it possible to replicate the back-end to different machines and have a scaled out solution? This is essentially multiple-end points with a single replicated database.

I could not find any information on this setup in the documentation. Is this supported and/or is there any information about setting up this scenario I should be aware of?

I should note that the remote sites are behind satellite links so very remote and cannot guarantee internet access availability. Hence why we distribute the server so they can do required data entry and replication happens when there is an up-link available.

Thanks in advance for any information/assistance that can be given.

**Comment by leastprivilege** [comment]
Yes, absolutely.

An IdentityServer is just a normal web application with our middleware for protocol support.

**Comment by Wayne-Mather** [comment]
Thanks for the information. That is what my take was, IdentityServer4 it is just essentially middl-ware for authentication that can use an EF store for persistent storage. I just was hoping security was similar to asp.net authentication where the users key is stored in the db.

**Comment by leastprivilege** [comment]
> IdentityServer4 it is just essentially middl-ware for authentication

No - IdentityServer is the protocol implementation (validating incoming protocol requests, producing responses).

Authentication is done by yourself, e.g. using ASP.NET Identity or any other user mgmt library that you prefer.

**Comment by Wayne-Mather** [comment]
Ok, thanks for the clarification. Guess I have more reading to do and playing with it to fully understand how to slip it into my current environment.
